`make check-helm-dependencies` is broken

```
./scripts/helm-dependencies.sh validate
=> vector-shared
./scripts/helm-dependencies.sh: line 60: DEPENDENCIES[@]: unbound variable
make: *** [check-helm-dependencies] Error 1

```

--- a/scripts/helm-dependencies.sh
+++ b/scripts/helm-dependencies.sh
@@ -41,8 +41,7 @@ update() {
     rm -rf "$CHART_VENDORED_DEPENDENCIES_PATH"
     mkdir -p "$CHART_VENDORED_DEPENDENCIES_PATH"
 
-    mapfile -t DEPENDENCIES < <(list-chart-dependencies "$CHART_PATH")
-    for DEPENDENCY_PAIR in "${DEPENDENCIES[@]}"; do
+    while IFS= read -r DEPENDENCY_PAIR; do
       read -ra KV <<<"$DEPENDENCY_PAIR"
       DEPENDENCY_NAME="${KV[0]}"
       DEPENDENCY_PATH="${KV[1]}"
@@ -52,7 +51,7 @@ update() {
 
       echo "Symlinking \"$DEPENDENCY_NAME\" with name \"$LINK_NAME\" and target \"$LINK_TARGET\"..."
       ln -sfn -T "$LINK_TARGET" "$LINK_NAME"
-    done
+    done < <(list-chart-dependencies "$CHART_PATH")
   done
 }
 
@@ -63,8 +62,7 @@ validate() {
     CHART_PATH="distribution/helm/$CHART"
     CHART_VENDORED_DEPENDENCIES_PATH="$CHART_PATH/charts"
 
-    mapfile -t DEPENDENCIES < <(list-chart-dependencies "$CHART_PATH")
-    for DEPENDENCY_PAIR in "${DEPENDENCIES[@]}"; do
+    while IFS= read -r DEPENDENCY_PAIR; do
       read -ra KV <<<"$DEPENDENCY_PAIR"
       DEPENDENCY_NAME="${KV[0]}"
       DEPENDENCY_PATH="${KV[1]}"
@@ -74,7 +72,7 @@ validate() {
 
       echo "Validating \"$DEPENDENCY_NAME\" at \"$VENDORED_PATH\" against \"$UPSTREAM_PATH\"..."
       diff -qr "$VENDORED_PATH" "$UPSTREAM_PATH"
-    done
+    done < <(list-chart-dependencies "$CHART_PATH")
   done
 }
 

--- a/scripts/patch-packaged-helm-chart-versions.sh
+++ b/scripts/patch-packaged-helm-chart-versions.sh
@@ -30,8 +30,7 @@ TEMPDIR="$(mktemp -d)"
 tar -xf "$PACKAGED_CHART_FILE" --directory "$TEMPDIR"
 pushd "$TEMPDIR" >/dev/null
 
-mapfile -t CHART_FILES < <(find "$CHART_NAME" -mindepth 2 -name Chart.yaml )
-for CHART_FILE in "${CHART_FILES[@]}"; do
+while IFS= read -r CHART_FILE; do
   echo "=> Patching versions at $CHART_FILE"
 
   {
@@ -39,7 +38,7 @@ for CHART_FILE in "${CHART_FILES[@]}"; do
     | sed 's/^appVersion: .*$/appVersion: '"$APP_VERSION"'/g'
   } < "$CHART_FILE" > "$CHART_FILE.new"
   mv "$CHART_FILE.new" "$CHART_FILE"
-done
+done < <(find "$CHART_NAME" -mindepth 2 -name Chart.yaml )
 
 popd >/dev/null
 tar -czf "$PACKAGED_CHART_FILE" -C "$TEMPDIR" -T - <<< "$STRUCTURE_BEFORE"

